src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/EnvironmentVariablesScenario.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System;
+using System.Collections;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios
+{
+    /// <summary>
+    /// Prints out the environment variables.
+    /// </summary>
+    internal class EnvironmentVariablesScenario
+    {
+        public static Command Command()
+        {
+            Command command = new(TestAppScenarios.EnvironmentVariables.Name);
+            command.Handler = CommandHandler.Create((Func<CancellationToken, Task<int>>)ExecuteAsync);
+            return command;
+        }
+
+        public static Task<int> ExecuteAsync(CancellationToken token)
+        {
+            string[] acceptableCommands = new string[]
+            {
+                TestAppScenarios.EnvironmentVariables.Commands.PrintVars,
+                TestAppScenarios.EnvironmentVariables.Commands.IncVar,
+                TestAppScenarios.EnvironmentVariables.Commands.ShutdownScenario,
+            };
+            return ScenarioHelpers.RunScenarioAsync(async logger =>
+            {
+                while (true)
+                {
+                    string command = await ScenarioHelpers.WaitForCommandAsync(acceptableCommands, logger);
+
+                    switch (command)
+                    {
+                        case TestAppScenarios.EnvironmentVariables.Commands.PrintVars:
+                            IDictionary vars = Environment.GetEnvironmentVariables();
```

Comment:
Because the facility to process these log messages is part of the AppRunner, I think this should move to something that is callable by all tests, such as the ScenarioHelpers class; something like `LogAllEnvironmentVariables()`.

Reply:
Moving this to the `ScenarioHelper` would entail inserting this into the `WaitForCommandAsync` handler and simply allowing all tests to call `PrintVars` whenever they want let it be an app level command.

Reply:
That would be one step further than I was suggesting, which I think I would be okay with, although that means you would be forced to wait for some other command in the test app if all you wanted to do was send the environment variables.

Reply:
Alright, I did that. That's basically what the `EnvironmentVariable` scenario already does. It has it's own `ShutdownScenario` command to turn it off.